src/test/java/testsuite/integration/container/AuroraMySqlIntegrationTest.java

```diff
@@ -0,0 +1,596 @@
+          // Ignore, stop the thread
+        }
+      });
+      try (final Connection conn = connectToInstance(MYSQL_INSTANCE_1_URL + PROXIED_DOMAIN_NAME_SUFFIX, MYSQL_PROXY_PORT, props);
+          final Statement statement = conn.createStatement()) {
+        hasFailConnect = false;
+
+        // Execute long query
+        try {
+          final String QUERY = "select sleep(600)"; // 600s -> 10min
+          thread.start();
+          final ResultSet result = statement.executeQuery(QUERY);
+          fail("Sleep query finished, should not be possible with network downed.");
+        } catch (Exception throwables) { // Catching executing query
+          // Calculate and add detection time
+          final int failureTime = (int)(System.currentTimeMillis() - downtime.get());
+          avg.add(failureTime);
+          min = Math.min(min, failureTime);
+          max = Math.max(max, failureTime);
+        }
+      } catch (Exception throwables) { // Catching Connection connect & Statement creations
+        if (hasFailConnect) {
+          // Already failed trying to connect twice in a row
+          fail("Exception occurred twice when trying to connect or create statement");
+        }
+        i--; // Retry
+        hasFailConnect = true;
+      } finally {
+        thread.interrupt(); // Ensure thread has stopped running
+        try {
+          proxyInstance_1.toxics().get("DOWN-STREAM").remove();
+          proxyInstance_1.toxics().get("UP-STREAM").remove();
+        } catch (Exception e) {
+          // Ignore as toxics were never set
+        }
+      }
+    }
+
+    final Integer[] arr = {detectionTime, detectionInterval, detectionCount, sleepDelayMS, min, max, (int)avg.stream().mapToInt(a -> a).summaryStatistics().getAverage()};
+    final List<Integer> data = new ArrayList<>(Arrays.asList(arr));
+    dataList.add(data);
+  }
+
+  private static Stream<Arguments> generateFailureDetectionTimeParams() {
+    // detectionTime, detectionInterval, detectionCount, sleepDelayMS
+    return Stream.of(
+        Arguments.of(30000, 5000, 3, 5000), // Defaults
+        Arguments.of(6000, 1000, 1, 5000) // Aggressive
+    );
+  }
+
+  @Test
+  public void test_LostConnectionToWriter() throws SQLException, IOException {
+    final Properties props = initDefaultProps();
+    props.setProperty(PropertyKey.failoverTimeoutMs.getKeyName(), "10000");
+
+    // Connect to cluster
+    try (final Connection testConnection = connectToInstance(MYSQL_CLUSTER_URL + PROXIED_DOMAIN_NAME_SUFFIX, MYSQL_PROXY_PORT, props)) {
+      // Get writer
+      currWriter = selectSingleRow(testConnection, QUERY_FOR_INSTANCE);
+
+      // Put cluster & writer down
+      final Proxy proxyInstance = proxyMap.get(currWriter);
+      if (proxyInstance != null) {
+        proxyInstance.toxics().bandwidth("DOWN-STREAM", ToxicDirection.DOWNSTREAM, 0);
+        proxyInstance.toxics().bandwidth("UP-STREAM", ToxicDirection.UPSTREAM, 0);
```

Comment:
Can we put these two lines into a method. The method will only have 2 lines but we always use these two lines together. A descriptive method name will also help to explain what these lines do.

Reply:
Agree. I have similar suggestion.